server/service/integration_core_test.go

```diff
@@ -3511,6 +3511,69 @@ func (s *integrationTestSuite) TestPasswordReset() {
 	res.Body.Close()
 }
 
+func (s *integrationTestSuite) TestDeviceAuthenticatedEndpoints() {
+	t := s.T()
+
+	hosts := s.createHosts(t)
+
+	// create an auth token for hosts[0]
+	token := "much_valid"
+	mysql.ExecAdhocSQL(t, s.ds, func(db sqlx.ExtContext) error {
+		_, err := db.ExecContext(context.Background(), `INSERT INTO host_device_auth (host_id, token) VALUES (?, ?)`, hosts[0].ID, token)
+		return err
+	})
+
+	// get host without token
+	res := s.DoRawNoAuth("GET", "/api/v1/fleet/device/", nil, http.StatusNotFound)
+	res.Body.Close()
+
+	// get host with invalid token
+	res = s.DoRawNoAuth("GET", "/api/v1/fleet/device/no_such_token", nil, http.StatusUnauthorized)
+	res.Body.Close()
+
+	// get host with valid token
+	var getHostResp getHostResponse
+	res = s.DoRawNoAuth("GET", "/api/v1/fleet/device/"+token, nil, http.StatusOK)
```

Comment:
> same behavior as GET /api/latest/fleet/hosts/<host id> but with device token authentication

Maybe also check that the returned host here is the same as the one returned by `GET /api/latest/fleet/hosts/<host id>`.

(Similar with the other added endpoints.)

Reply:
There's already that check with the response: `require.Equal(t, hosts[0].ID, getHostResp.Host.ID)`, not sure what adding the other call would add to the test?

Reply:
:+1: Updated with those tests.